bpfman/src/types.rs

```diff
@@ -510,16 +1583,32 @@ impl ProgramData {
     // since the TC and XDP programs loaded by bpfman will have a ProgramType::Ext
     // rather than ProgramType::Xdp or ProgramType::Tc.
     // Kind should only be set on programs loaded by bpfman.
-    fn set_kind(&mut self, kind: ProgramType) -> Result<(), BpfmanError> {
-        sled_insert(
-            &self.db_tree,
-            KIND,
-            &(Into::<u32>::into(kind)).to_ne_bytes(),
-        )
+    fn set_kind(&mut self, kind: BpfProgType) -> Result<(), BpfmanError> {
+        sled_insert(&self.0, KIND, &(Into::<u32>::into(kind)).to_ne_bytes())
+    }
+
+    fn set_is_tcx(&mut self, v: bool) -> Result<(), BpfmanError> {
```

Comment:
This is better than the previous hack, but IMO, it would be even less hacky if we maintained the actual program type for all the programs (i.e., uprobe, kprobe, tc, tcx, etc.).  Then, for specific cases where we need the kernel program type, it's easy to convert.  I'm not asking you to do anything about this right now -- it just bothers me whenever I see the complications of the decision to use the kernel types as we do.

Reply:
#1429